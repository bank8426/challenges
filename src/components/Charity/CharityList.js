import React,{useState} from 'react'
import styled from 'styled-components';
import Card from './Card';
import PropTypes from 'prop-types'

/**div to display all Charity Card components */
const StyledCharityList = styled.div`
  /**display in grid with 2 elements per row */
  display: grid;
  grid-template-columns: repeat(2,1fr);
  justify-items: center;

  /**display one card in small screen size*/
  @media (max-width: 768px) {
    display: block;
  }
`;

/**
 * div to display all charities,
 * handle which charity currently selected to make that charity's CardPayment appear
 */
const CharityList = ({charities}) => {
  /** hook for set setSelectCharity*/
  const [selectedCharity,setSelectCharity] = useState(0)
  return (
    <StyledCharityList>
      {
        charities.map((item, i) => 
          <Card key={i} 
            item={item} 
            selectedCharity={selectedCharity} 
            setSelectCharity={setSelectCharity} 
            index={i}/>)
      }
    </StyledCharityList>
  )
}

CharityList.defaultProps ={
  charities : [],
}

CharityList.propTypes={
  /** array of charities to be display.*/
  charities: PropTypes.arrayOf(PropTypes.shape({
    /** id of charity*/
    id :PropTypes.number.isRequired,
    /** name string of charity*/
    name:PropTypes.string.isRequired,
    /** image url of charity*/
    image:PropTypes.string.isRequired,
    /** acceptance currency of charity*/
    currency:PropTypes.string.isRequired,
  })),
}

export default CharityList
