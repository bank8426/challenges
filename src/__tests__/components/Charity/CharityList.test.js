import React from 'react'
import CharityList from '../../../components/Charity/CharityList'
import renderer from 'react-test-renderer'
import {mount} from 'enzyme'
import 'jest-styled-components'
import {charities} from '../../../../tools/mockData'

it('it should display CharityList', () => {
  const tree = renderer.create(<CharityList />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('should display charities in CharityList', () => {
  const initCharities =  [charities[0]]
  const tree = renderer.create(<CharityList charities={initCharities} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('should display 2 Cards in CharityList', () => {
  const initCharities =  [charities[0],charities[1]]
  const wrapper = mount(<CharityList charities={initCharities} />)
  expect(wrapper.find('Card').length).toEqual(2)
})