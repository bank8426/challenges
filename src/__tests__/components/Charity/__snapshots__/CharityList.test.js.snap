// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should display CharityList 1`] = `
.c0 {
  display: grid;
  grid-template-columns: repeat(2,1fr);
  justify-items: center;
}

@media (max-width:768px) {
  .c0 {
    display: block;
  }
}

<div
  className="c0"
/>
`;

exports[`should display charities in CharityList 1`] = `
.c1 {
  margin: 0 10px 50px;
  background: white;
  box-shadow: 2px 2px 10px rgba(0,0,0,0.25);
  border-radius: 10px;
  top: 10px;
  position: relative;
  opacity: 0;
  -webkit-animation: cMHBRe 3s forwards cubic-bezier(0.2,0.8,0.2,1) 0.02s;
  -webkit-animation: cMHBRe 3s forwards cubic-bezier(0.2,0.8,0.2,1) 0.02s;
  animation: cMHBRe 3s forwards cubic-bezier(0.2,0.8,0.2,1) 0.02s;
}

.c2 {
  width: 650px;
  overflow: hidden;
  background-image: url(undefinedbaan-kru-noi.jpg);
  height: 300px;
  border-radius: 10px 10px 0px 0px;
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
}

.c4 {
  margin: 1em 0em 1em 1em;
}

.c5 {
  margin: auto 1em auto 0;
}

.c3 {
  display: grid;
  grid-template-columns: 1fr max-content;
}

.c0 {
  display: grid;
  grid-template-columns: repeat(2,1fr);
  justify-items: center;
}

@media (max-width:1366px) {
  .c2 {
    width: 600px;
    height: 275px;
  }
}

@media (max-width:1280px) {
  .c2 {
    width: 480px;
    height: 230px;
  }
}

@media (max-width:1024px) {
  .c2 {
    width: 385px;
    height: 195px;
  }
}

@media (max-width:834px) {
  .c2 {
    width: 345px;
    height: 175px;
  }
}

@media (max-width:768px) {
  .c2 {
    width: 100%;
    height: 300px;
  }
}

@media (max-width:500px) {
  .c2 {
    width: 100%;
    height: 250px;
  }
}

@media (max-width:834px) and (min-width:768px) {
  .c5 {
    margin: 0.5em 0.5em 0.5em 0;
  }
}

@media (max-width:768px) {
  .c0 {
    display: block;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      src="undefinedbaan-kru-noi.jpg"
    />
    <div
      className="c3"
    >
      <h3
        className="c4"
      >
        Baan Kru Noi
      </h3>
      <button
        className="primary c5"
        onClick={[Function]}
      >
        Donate
      </button>
    </div>
  </div>
</div>
`;
